invenio_app_ils/ui/src/invenio_app_ils/common/api/locations/internalLocation.js

```diff
@@ -3,34 +3,23 @@ import { internalLocationSerializer as serializer } from './serializer';
 
 const internalLocationURL = '/internal-locations/';
 
-const get = internalLocationPid => {
-  return http
-    .get(`${internalLocationPid}${internalLocationPid}`)
-    .then(response => {
-      response.data = serializer.fromJSON(response.data);
-      return response;
-    });
+const get = async internalLocationPid => {
+  const response = await http.get(
+    `${internalLocationURL}${internalLocationPid}`
+  );
+  response.data = serializer.fromJSON(response.data);
+  return response;
 };
 
-const list = query => {
-  if (query) {
-    return http.get(`${internalLocationURL}?q=${query}`).then(response => {
-      response.data.total = response.data.hits.total;
-      response.data.hits = response.data.hits.hits.map(hit =>
-        serializer.fromJSON(hit)
-      );
-      return response;
-    });
-  } else {
-    return http.get(`${internalLocationURL}`).then(response => {
-      response.data.total = response.data.hits.total;
-      response.data.hits = response.data.hits.hits.map(hit =>
-        serializer.fromJSON(hit)
-      );
-      return response;
-    });
-  }
+const list = async (query = '') => {
```

Comment:
we should discuss if we prefer to use `await` - `async` or we stay with using `Promise` as we should keep the code consistent. 
Maybe this will help with the discussion: https://medium.com/limitedio/should-i-use-promises-or-async-await-126ab5c98789

Reply:
await/async! :)